src/content/en/updates/posts/2016/05/houdini.md

```diff
@@ -0,0 +1,366 @@
+Microsoft, HP, Intel and Google working together to expose certain parts of the
+CSS engine to web developers. The task force is working on a *collection of
+drafts* with the goal to get them accepted by the W3C to become actual web
+standards. They set themselves a few high-level goals, turned them into
+specification drafts which in turn gave birth to a set of supporting,
+lower-level specification drafts. The combination of these drafts is what is
+usually meant when someone is talking about “Houdini”. At the time of writing,
+the [list of drafts][Houdini Drafts] is still incomplete and some of the drafts
+are mere placeholders. That’s how early in development of Houdini we are.
+
+<iframe width="100%" height="315" src="https://www.youtube.com/embed/EUlIxr8mk7s" frameborder="0" allowfullscreen></iframe>
+
+*Disclaimer:* I want to give a quick overview of the Houdini drafts so you have
+*an idea of what kind of problems Houdini tries to tackle. As far as the current
+*state of the specs allow, I’ll try to give code examples, as well. Keeping that
+*in mind, please be aware that all of these specs are *drafts* and very
+*volatile. There’s no guarantee that these code samples are even remotely
+*correct in the future or that any of these drafts become reality at all.
+
+## The specifications
+### Worklets ([spec][Worklets spec])
+
+Worklets by themselves are not really useful. They are a concept introduced to
+make many of the later drafts possible. If you thought of Web Workers when you
+read “worklet”, you are not wrong. They have a lot of conceptual overlap. So why
+a new thing when we already have workers? Houdini’s goal is to expose new APIs
+to allow web developers to hook up their own code into the CSS engine and the
+surrounding systems. It’s probably not unrealistic to assume that some of these
+code fragments will have to be run *every. single. frame*. Some of them have to
+by definition. Quoting the [Web Worker spec]:
+
+> Workers [...] are relatively heavy-weight, and are not intended to be used in
+large numbers. For example, it would be inappropriate to launch one worker for
+each pixel of a four megapixel image.
+
+That means web workers are not viable for the things Houdini plans to do.
+Therefore, worklets were invented. Worklets make use of ES2015 classes as nice
+way to define a collection of methods, the signatures of which are predefined by
+the type of the worklet. They are light-weight and short-lived.
+
+### Paint Worklet ([spec][Paint Worklet spec])
+
+I am starting with this as it is introduces the fewest new concepts. From the
+spec draft itself:
+
+> The paint stage of CSS is responsible for painting the background, content and
+highlight of an element based on that element’s geometry (as generated by the
+layout stage) and computed style.
+
+This allows you not only to define how an element should draw itself (think of
+the synergy with Web Components!) but also *alter* the visuals of existing
```

Comment:
Not everyone will know what Web Components are. Link to http://webcomponents.org/ or some other definition.